core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/lru/LruBlockCache.java

```diff
@@ -674,9 +678,9 @@ public long heapSize() {
   }
 
   public static long calculateOverhead(long maxSize, long blockSize, int concurrency) {
-    return CACHE_FIXED_OVERHEAD + ClassSize.CONCURRENT_HASHMAP
-        + ((int) Math.ceil(maxSize * 1.2 / blockSize) * ClassSize.CONCURRENT_HASHMAP_ENTRY)
-        + (concurrency * ClassSize.CONCURRENT_HASHMAP_SEGMENT);
+    long entryPart = Math.round(maxSize * 1.2 / blockSize) * CONCURRENT_HASHMAP_ENTRY;
+    long segmentPart = (long) concurrency * CONCURRENT_HASHMAP_SEGMENT;
+    return CACHE_FIXED_OVERHEAD + CONCURRENT_HASHMAP + entryPart + segmentPart;
```

Comment:
I broke this calculation into multiple lines to make it easier to read/understand.  I then replaced the `Math.ceil()` with `Math.round()` since it returns the proper type that we need.  All of the `ClassSize.CONCURRENT` constants are small int types but `concurrency` could be larger so I casted that to a long.